### DISCUSSION TITLE
Starting speed before acceleration

### DISCUSSION THREAD

**Discussion by HeldeReis** [gin66/FastAccelStepper#153] [Ideas]
Hello,

I came across your library when my application (rotating a mass 45°) started missing steps. During the project development the mass increased and I was not able to change the motors.
Your library behaved wonderfully helping me to keep the rotation of the mass as fast as I needed without missing steps, but I believe it could benefit from having a fixed starting/stopping speed. The idea is to start rotating the motor at a fixed speed (user configurable) and then generate the ramp as is done right now. This would allow (in applications like mine) to achieve the maximum rotating speed in the minimum time.

**Comment by gin66** [comment]
Could you please give it a try in your project ?

At start of a ramp (so from standstill), the speed is regulated by the performed steps aka s on the ramp:
https://github.com/gin66/FastAccelStepper/blob/86556d8b514a67d4d60fd6c9b4f694fd9ec2e4f1/src/RampConstAcceleration.h#L162

Please change the value of 0 to any value greater than 0. I expect, the ramp generator then jumps to the related speed and then continues to accelerate. Perhaps this works.

That value s is linked to speed via `s = v²/2a`. So your start speed should then be `v = sqrt(2 * a * s)` with the chosen value s being greater than 0. That value is independent of the direction, so only positive values make sense.

**Reply by gin66** [reply]
Increase of stop speed is a way more complicated, so please try with start firstly. Besides, if the app knows, that stop is upcoming, the app can increase the acceleration value (which is then for deceleration ), which provides a faster stop.

**Comment by gin66** [comment]
btw: if shortest time to max. speed is required, then a linear speed ramp does not fit.

**Comment by gin66** [comment]
The upcoming 0.29.2 implements a jump start feature, which allows to jump on any point on the ramp from standstill. As every point in the ramp relates to a speed, the start speed can be controlled in this way.

**Reply by HeldeReis** [reply]
> The upcoming 0.29.2 implements a jump start feature, which allows to jump on any point on the ramp from standstill. As every point in the ramp relates to a speed, the start speed can be controlled in this way.

Thank you gin66

Right now i don't have the hardware to test it anymore, but when i have i will report it here.
